receiver/receiverhelper/scrapercontroller.go

```diff
@@ -180,26 +183,21 @@ func (sc *scraperController) startScraping() {
 // Scrapers, records observability information, and passes the scraped metrics
 // to the next component.
 func (sc *scraperController) scrapeMetricsAndReport(ctx context.Context) {
-	// TODO: add observability metrics support
-	var errs []error
+	ctx = obsreport.ReceiverContext(ctx, sc.name, "", "")
 
 	metrics := pdata.NewMetrics()
 
 	for _, rms := range sc.resourceMetricScrapers {
-		resourceMetrics, err := rms.Scrape(ctx)
-		if err != nil {
-			errs = append(errs, err) // TODO: handle partial errors
-		}
-
+		// ignore errors while scraping
+		resourceMetrics, _ := rms.Scrape(ctx)
 		resourceMetrics.MoveAndAppendTo(metrics.ResourceMetrics())
 	}
 
-	// TODO: report error
-	if len(errs) > 0 {
-		componenterror.CombineErrors(errs)
-	}
+	metricCount, dataPointCount := metrics.MetricAndDataPointCount()
 
-	sc.nextConsumer.ConsumeMetrics(ctx, metrics)
+	ctx = obsreport.StartMetricsReceiveOp(ctx, sc.name, "")
+	err := sc.nextConsumer.ConsumeMetrics(ctx, metrics)
+	obsreport.EndMetricsReceiveOp(ctx, "", metricCount, dataPointCount, err)
```

Comment:
@bogdandrutu / @jrcamp - does this look roughly right in terms of the observability info that we would want for scrapers?

i.e. Record:

- Traces / Metrics per Scraper (reports errors scraping metrics)
- Traces / Metrics for the Receiver itself (reports errors pushing all scraped metrics into pipeline)

Reply:
> Traces / Metrics per Scraper (reports errors scraping metrics)

Are the metrics dimensionalized per-scraper instance?

> Traces / Metrics for the Receiver itself (reports errors pushing all scraped metrics into pipeline)

Could this be made part of the non-scraping part of receiverhelper? Looks like exporterhelper does this, not sure why receiver helper doesn't (@bogdandrutu). I forget, does the scraping create its own receiver or does it rely on the other parts of receiverhelper?

Reply:
The only functions in the receiver interface are `Start` & `End`, so having helpers for this doesn't really achieve much. "Push" based receivers generally set up a handler(s) of some sort where this code would live. We could make a "push" based receiver helper at some point automatically creates HTTP and/or gRPC handlers, but this would need significant thought.

Initially I'd kept the base receiver code separate with this future extension in mind, but Bogdan consolidated when refactoring. I think it makes sense to keep this simple for now as is. We can always refactor in the future if we want to add other types of receiverhelpers without changing the interface here.